scene/2d/node_2d.cpp

```diff
@@ -257,25 +257,45 @@ real_t Node2D::get_global_rotation() const {
 	return get_global_transform().get_rotation();
 }
 
-void Node2D::set_global_rotation(real_t p_radians) {
+real_t Node2D::get_global_skew() const {
+	return get_global_transform().get_skew();
+}
+
+void Node2D::set_global_rotation(const real_t p_radians) {
 	CanvasItem *pi = get_parent_item();
 	if (pi) {
-		const real_t parent_global_rot = pi->get_global_transform().get_rotation();
-		set_rotation(p_radians - parent_global_rot);
+		Transform2D new_transform = get_global_transform();
+		new_transform.set_rotation(p_radians);
+		new_transform = pi->get_global_transform().affine_inverse() * new_transform;
+		set_rotation(new_transform.get_rotation());
 	} else {
 		set_rotation(p_radians);
 	}
 }
 
+void Node2D::set_global_skew(const real_t p_radians) {
+	CanvasItem *pi = get_parent_item();
+	if (pi) {
+		Transform2D new_transform = get_global_transform();
+		new_transform.set_skew(p_radians);
+		new_transform = pi->get_global_transform().affine_inverse() * new_transform;
```

Comment:
It seems that GitHub's syntax highlighting doesn't like `pi`, we should change it to `parent`:

```suggestion
	CanvasItem *parent = get_parent_item();
	if (parent) {
		Transform2D new_transform = get_global_transform();
		new_transform.set_skew(p_radians);
		new_transform = parent->get_global_transform().affine_inverse() * new_transform;
```